common/src/services/crypto.service.ts

```diff
@@ -3,81 +3,71 @@ import * as bigInt from 'big-integer';
 import { EncryptionType } from '../enums/encryptionType';
 import { HashPurpose } from '../enums/hashPurpose';
 import { KdfType } from '../enums/kdfType';
+import { StorageKey } from '../enums/storageKey';
 
 import { EncArrayBuffer } from '../models/domain/encArrayBuffer';
 import { EncryptedObject } from '../models/domain/encryptedObject';
 import { EncString } from '../models/domain/encString';
+import { KeySuffixOptions, SettingStorageOptions } from '../models/domain/settingStorageOptions';
 import { SymmetricCryptoKey } from '../models/domain/symmetricCryptoKey';
-import { ProfileOrganizationResponse } from '../models/response/profileOrganizationResponse';
 
+
+import { AccountService } from '../abstractions/account.service';
 import { CryptoService as CryptoServiceAbstraction } from '../abstractions/crypto.service';
 import { CryptoFunctionService } from '../abstractions/cryptoFunction.service';
 import { LogService } from '../abstractions/log.service';
 import { PlatformUtilsService } from '../abstractions/platformUtils.service';
-import {
-    KeySuffixOptions,
-    StorageService,
-} from '../abstractions/storage.service';
-
-import { ConstantsService } from './constants.service';
 
 import { sequentialize } from '../misc/sequentialize';
 import { Utils } from '../misc/utils';
 import { EEFLongWordList } from '../misc/wordlist';
+
+import { ProfileOrganizationResponse } from '../models/response/profileOrganizationResponse';
 import { ProfileProviderOrganizationResponse } from '../models/response/profileProviderOrganizationResponse';
 import { ProfileProviderResponse } from '../models/response/profileProviderResponse';
 
-export const Keys = {
-    key: 'key', // Master Key
-    encOrgKeys: 'encOrgKeys',
-    encProviderKeys: 'encProviderKeys',
-    encPrivateKey: 'encPrivateKey',
-    encKey: 'encKey', // Generated Symmetric Key
-    keyHash: 'keyHash',
-};
 
 export class CryptoService implements CryptoServiceAbstraction {
-    private key: SymmetricCryptoKey;
-    private encKey: SymmetricCryptoKey;
-    private legacyEtmKey: SymmetricCryptoKey;
-    private keyHash: string;
-    private publicKey: ArrayBuffer;
-    private privateKey: ArrayBuffer;
-    private orgKeys: Map<string, SymmetricCryptoKey>;
-    private providerKeys: Map<string, SymmetricCryptoKey>;
-
-    constructor(private storageService: StorageService, protected secureStorageService: StorageService,
-        private cryptoFunctionService: CryptoFunctionService, protected platformUtilService: PlatformUtilsService,
-        protected logService: LogService) {
+    constructor(private cryptoFunctionService: CryptoFunctionService, protected platformUtilService: PlatformUtilsService,
+        protected logService: LogService, protected accountService: AccountService) {
     }
 
     async setKey(key: SymmetricCryptoKey): Promise<any> {
-        this.key = key;
-
-        await this.storeKey(key);
```

Comment:
Thanks for pointing this out. Could you explain why there is an override in electron? It looks like the only thing different in the electron function is that it includes the key suffix in the storage key, but the default implementation of the method still requires 'auto' or 'biometric' to be true, so couldn't we just send the key suffix no matter what and remove the override?

Reply:
To be honest, I haven't looked into that area much, I know @MGibson1 worked on it. If they behave the same it should be fine to get rid of it, but we currently implement the `storeKey` for the electron crypto service, so that function should in that case be removed.

Reply:
@MGibson1 if you check this out and can advise please do. I didn't see any harm implementing the electronService override in place of the default cryptoService method, and have done so.